Tests/ApolloTests/StarWars/API.swift

```diff
@@ -635,104 +760,130 @@ public final class HeroParentTypeDependentFieldQuery: GraphQLQuery {
   public struct Data: GraphQLMappable {
     public let hero: Hero?
 
-    public init(reader: GraphQLResultReader) throws {
-      hero = try reader.optionalValue(for: Field(responseName: "hero", arguments: ["episode": reader.variables["episode"]]))
+    public init(values: [Any?]) {
+      hero = values[0] as! Hero?
     }
 
     public struct Hero: GraphQLMappable {
       public let __typename: String
       public let name: String
-
       public let asHuman: AsHuman?
       public let asDroid: AsDroid?
 
-      public init(reader: GraphQLResultReader) throws {
-        __typename = try reader.value(for: Field(responseName: "__typename"))
-        name = try reader.value(for: Field(responseName: "name"))
-
-        asHuman = try AsHuman(reader: reader, ifTypeMatches: __typename)
-        asDroid = try AsDroid(reader: reader, ifTypeMatches: __typename)
+      public init(values: [Any?]) {
+        __typename = values[0] as! String
+        name = values[1] as! String
+        asHuman = values[2] as! AsHuman?
+        asDroid = values[3] as! AsDroid?
       }
 
-      public struct AsHuman: GraphQLConditionalFragment {
+      public struct AsHuman: GraphQLFragment {
         public static let possibleTypes = ["Human"]
 
+        public static let selectionSet: [Selection] = [
+          Field("__typename", type: .nonNull(.scalar(String.self))),
+          Field("name", type: .nonNull(.scalar(String.self))),
+          Field("friends", type: .list(.object(AsHuman.Friend.self)), selectionSet: [
+            Field("__typename", type: .nonNull(.scalar(String.self))),
+            Field("name", type: .nonNull(.scalar(String.self))),
+            FragmentSpread(AsHuman.Friend.AsHuman.self),
+          ]),
+        ]
+
         public let __typename: String
         public let name: String
         public let friends: [Friend?]?
 
-        public init(reader: GraphQLResultReader) throws {
-          __typename = try reader.value(for: Field(responseName: "__typename"))
-          name = try reader.value(for: Field(responseName: "name"))
-          friends = try reader.optionalList(for: Field(responseName: "friends"))
+        public init(values: [Any?]) {
+          __typename = values[0] as! String
+          name = values[1] as! String
+          friends = values[2] as! [Friend?]?
         }
 
         public struct Friend: GraphQLMappable {
```

Comment:
Yes, we singularize `friends` to `Friend`. There is the potential for collision, so I have been worried about that. I preferred clean type names though, because they do show up in autocompletion, and you need to specify them when storing values.